### DISCUSSION TITLE
Help getting started with Joomla/Astroid in MAMP Pro

### DISCUSSION THREAD

**Discussion by paurray** [templaza/astroid-framework#307] [Q&A]
Hi

(I want to test some stuff for creating Astroid Tutorials so it is in a round about way related to Astroid)

For better or worse I bought and used MAMP Pro a couple of years ago.
https://www.mamp.info/en/mac/
And felt reasonably comfortable in it testing stuff in a Joomla installation.
Fast forward to today and I have MAMP Pro installed on a new computer.
I forgot pretty much everything I knew about MAMP.
And I am really lost just even getting a Joomla install up and running…

This is what I am trying to follow along with:
https://documentation.mamp.info/en/MAMP-Mac/FAQ/How-do-I-install-Joomla/
Maybe I am missing something but I just do not get the instructions.

Most videos, instructions that I have found so far seem outdated or incomplete!

Could some one point me in the right direction?

Thanks for any help

Paul

**Comment by Chacapamac** [comment]
I @paurray , I will try to help!
You right, trying to find a new decent installation/first Steps on Mamp seem to be really hard?????.

If you already use MAMP before I will start with the big changes for the newer versions.

• **The first big change** is that MAMP will now use your **Mac/Users/YourMacUserFolder/ Sites** (created by MAMP installation)**/localhost**
— compare to the old way using **Mac/Applications/MAMP/htdocs**
— When installing on a older Mamp version (using the old way) Mamp will move your old **Mac/Applications/MAMP/htdocs** content to the new **Mac/Users/YourMacUserFolder/ Sites** (created by MAMP installation)**/localhost**
_**Note:** I thing this move represent the right Localhost emplacement that newer Mac will use to create a Mac server_

![Screen Shot 2023-03-17 at 10 44 52 AM](https://user-images.githubusercontent.com/1907276/225944671-42121b67-0f3e-4b44-9bd6-d328c169c51b.jpg)

**Note:** Using my subfolder way, your path to your websites will become —> http://localhost/MainSubFolder/WebsiteFolder/(nothing for front-end or add administrator/ for administration

For the rest You use Mamp the same way.

Ask if you other questions....

**Good Day!**

**Comment by paurray** [comment]
Many thanks for the push in the right direction.
Ok I think I understand now.

This also helped me even if it is for Wordpress but same principles:
https://www.youtube.com/watch?v=10rsWKEDFvw

Broadly speaking it is all about:

- Getting the unzipped Joomla package in the right folder.
https://downloads.joomla.org/
- Which at least on a M1 Silicon Mac with MAMP Pro means in the Sites/localhost folder
- Then creating a new/custom host in MAMP Pro.
- Pointing the custom host at the folder to the correct path
Eg My User Name/Sites/localhost/joomla/joomla_FRESH
- Opening and starting the MAMP server

<img width="423" alt="image" src="https://user-images.githubusercontent.com/9163424/226388375-3aca1a4c-0147-4be6-a075-07d30f7c5e5b.png">

<img width="423" alt="image" src="https://user-images.githubusercontent.com/9163424/226388414-291d087d-a568-4832-8d4a-efe15f35b5d1.png">

- Having created a sub folder for the specific installation in User Name/Sites/localhost/joomla/
- On the finder level drag and drop a fresh Joomla instalation into EG User Name/Sites/localhost/joomla/joomla_FRESH

- Back in MAMP go to the Joomla install in your web browser as shown in the next screen shot.

<img width="423" alt="image" src="https://user-images.githubusercontent.com/9163424/226392486-b466b17f-2f8c-42a8-be55-6a78b7a849c1.png">

- Then it is the usual Joomla install

<img width="423" alt="image" src="https://user-images.githubusercontent.com/9163424/226392717-ec56bbb0-d6e3-4c02-8edf-52f824475fe0.png">

<img width="423" alt="image" src="https://user-images.githubusercontent.com/9163424/226392752-2c9bf8f9-0aed-40d7-bea6-7b7fac215de7.png">

<img width="423" alt="image" src="https://user-images.githubusercontent.com/9163424/226392772-a7aba925-90c0-4b21-b6dc-34a1968b040c.png">

- Was able to login and make a couple of very basic changes

<img width="423" alt="image" src="https://user-images.githubusercontent.com/9163424/226207451-f505b6ba-b0e2-45f0-b895-80ff6775f9fa.png">

Question :-)

I do however have one question still.

Lets say I am creating 7 lessons.
And I want a before the lesson and after lesson version.
Building on each other.
This way I can
- go through the whole thing once
- then multiple times on a step by step basis for screen recording!

EG

Lesson 01 Install Astroid (START HERE)
Lesson 01 Install Astroid (COMPLETED/Start for Next Lesson)
Lesson 02 Publish a custom module to the Home page with Joomla/Astroid (START HERE)
Lesson 02 Publish a custom module to the Home page with Joomla/Astroid (COMPLETED/Start for Next Lesson)
Lesson 03 Publish multiple custom modules to different site pages (START HERE)
Lesson 03 Publish multiple custom modules to different site pages (COMPLETED/Start for Next Lesson)

ETC

How would I do this?
I have played around with Snapshots (seem to be more about emergency back up)
And also tried cloning but end up with changes made in one clone showing up in another!

Another little push in the right direction would be greatfiully appreciated.

Hope this makes sense

Thanks

Paul

**Reply by Chacapamac** [reply]
@paurray 
Yes, your “Then creating a new/custom host in MAMP Pro.” for each website is the “Proper” way to go and give you more possibility on each website.
In my case I use the simple subfolder way for many years, just because its simple.

![Screen Shot 2023-03-21 at 2 04 30 PM](https://user-images.githubusercontent.com/1907276/226703643-7d904386-6224-42d2-88bb-283169b8bb5f.jpg)

> Question :-)
> 
> I do however have one question still.
> 
> Lets say I am creating 7 lessons.
> And I want a before the lesson and after lesson version.
> Building on each other.
> This way I can
> 
>     go through the whole thing once
>     then multiple times on a step by step basis for screen recording!
> 
> 

**I’m sorry, I lost you there, I’m not sure what you want to achieve here?**

**Comment by paurray** [comment]
@Chacapamac 
> I’m sorry, I lost you there, I’m not sure what you want to achieve here?

All good you have already been a huge help.
I will use this for screen recording:
https://www.telestream.net/screenflow/overview.htm
With a website versioning system I am more flexible.

EG I can:
- Build a site once from start to finish
- Save versions of it as I go along eg Step/Lesson 01, 02. 03, 04, 05,06,07 (finished) 
- If for eg I messed up in the middle of 6 and need to redo/tweak something I can just jump into version 6/step 6 of site.

Your answer helps me to do this thanks

best
Paul

**Reply by Chacapamac** [reply]
Cool @paurray 
Can I mark this discussion as “**SOLVED**”?

**Comment by paurray** [comment]
@Chacapamac 
Yes of course. Sorry my bad. Still finding my way around Github
thanks
Paul

**Reply by Chacapamac** [reply]
Cool, don’t worry it is just to keep up with the support.

**Good Day, Happy to Help!**
